tlspuffin/src/tls/seeds.rs

```diff
@@ -266,6 +267,49 @@ pub fn seed_successful(client: AgentName, server: AgentName) -> Trace<TlsQueryMa
     }
 }
 
+pub fn seed_successful_with_flights(
+    client: AgentName,
+    server: AgentName,
+) -> Trace<TlsQueryMatcher> {
+    Trace {
+        prior_traces: vec![],
+        descriptors: vec![
+            AgentDescriptor::new_client(client, TLSVersion::V1_3),
+            AgentDescriptor::new_server(server, TLSVersion::V1_3),
+        ],
+        steps: vec![
+            OutputAction::new_step(client),
+            // Client Hello Client -> Server
+            Step {
+                agent: server,
+                action: Action::Input(InputAction {
+                    recipe: term! {
+                        (client, 0)/MessageFlight<Message,OpaqueMessage>
+                    },
+                }),
+            },
+            // ServerHello/EncryptedExtensions/Certificate/CertificateVerify/ServerFinished -> Client
+            Step {
+                agent: client,
+                action: Action::Input(InputAction {
+                    recipe: term! {
+                        (server, 0)/MessageFlight<Message,OpaqueMessage>
+                    },
+                }),
+            },
+            // Client Finished -> server
+            Step {
+                agent: server,
+                action: Action::Input(InputAction {
+                    recipe: term! {
+                        (client, 1)/MessageFlight<Message,OpaqueMessage>
+                    },
+                }),
+            },
+        ],
+    }
+}
+
 /// Seed which triggers a MITM attack. It changes the cipher suite. This should fail.
 pub fn seed_successful_mitm(client: AgentName, server: AgentName) -> Trace<TlsQueryMatcher> {
```

Comment:
This is subsumed by the previous one, no?

Reply:
seed_successful_mitm modifies the client hello and forward the other messages of the handshake, while seed_successful just forward a complete handshake.
Is it normal that this trace is called seed_**successful**_mitm knowing that this trace should fail (cf. the comment above the trace) ?